tests/models/test_measurements.py

```diff
@@ -83,7 +84,40 @@ def test_measurements_attributes(measurer, ptf_datastore, test_config):
     m.flux_apertures[m.best_aperture] = original_flux
     new_im.zp.dzp = original_zp_err
 
-    # TODO: add test for limiting magnitude (issue #143)
+    #test limiting magnitude estimation
+    srcList = ptf_datastore.sources
+    #if interactive testing enabled, make and save a Magnitude vs SNR plot (for debugging)
+    makePlot = os.getenv('INTERACTIVE')
+    if makePlot:
+        limMagResults = srcList.estimate_lim_mag(aperture=1, givePlotParams=makePlot)
+        limMagEst = limMagResults[0]
+        snrs = limMagResults[1]
+        mags = limMagResults[2]
+        grad = limMagResults[3]
+        intercept = limMagResults[4]
+    else:
+        limMagEst = srcList.estimate_lim_mag(aperture=1)
+
+    #Skip the lim mag test if using PSF photometry (i.e. if limMagEst==None)
+    if limMagEst != None:
+        if makePlot: 
+            xdata = np.linspace(np.log(3),np.log(20),1000)
+            plt.plot(snrs,mags,linewidth=0,marker='o',c='midnightblue')
+            plt.plot(xdata, grad * xdata + intercept, c='firebrick')
+            plt.xlabel('log SNR')
+            plt.ylabel('magnitude')
+            plt.title('Limiting magntiude = {:.2f} mag'.format(limMagEst))
+            ymin,ymax = plt.gca().get_ylim()
+            plt.vlines(x=np.log(5),ymin=ymin,ymax=ymax)
+            plt.hlines(y=limMagEst,xmin=np.log(3),xmax=np.log(20))
+            plt.xlim(np.log(3),np.log(20))
+            plt.ylim(ymin,ymax)
+            plt.savefig('plots/snr_mag_plot.png')
+            plt.show()
```

Comment:
I would argue that what would be really useful is to add the plotting option to the `estimate_lim_mag` function. That means you can always get feedback on that fit, regardless of tests. 

I would also use the fixture `blocking_plots` in the `show()` function that follows after the call to `estimate_lim_mag`, as in `plt.show(block=blocking_plots)`. That would stop the test until you close the window. 

I would also say that regardless of blocking or not (i.e., regardless of interactive usage) you should always want to save this plot as an artefact of running this test. It doesn't hurt.